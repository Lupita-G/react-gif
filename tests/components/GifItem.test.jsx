import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas en <GifItem/>', () => { 

    const title = 'App';
    const url = 'https://hohola.com//app.jpg';

    test('debe de hacer match con el snapshot', () => { 
       const { container } = render(<GifItem title={ title } url={ url }/>)
       expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 

        render(<GifItem title={ title } url={ url }/>);
        //screen.debug();
        //expect( screen.getByRole('img').src).toBe(url);
        
        const { alt, src } = screen.getByRole('img');
        expect( alt ).toBe( alt );
        expect( src ).toBe( url);  
    });

    test('debe de mostrar el titulo', () => { 
        render(<GifItem title={ title } url={ url }/>);

        expect(screen.getByText(title)).toBeTruthy();
    })
 })