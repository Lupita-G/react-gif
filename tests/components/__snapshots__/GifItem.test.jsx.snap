// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="App"
      src="https://hohola.com//app.jpg"
    />
    <p>
       
      App
       
    </p>
  </div>
</div>
`;
